tests/test_scalars.py

```diff
@@ -0,0 +1,287 @@
+        pd.Interval,
+        pd.Timedelta,
+    )
+
+    check(assert_type(interval_i / 3, "pd.Interval[float]"), pd.Interval, float)
+    check(assert_type(interval_f / 3, "pd.Interval[float]"), pd.Interval, float)
+    check(
+        assert_type(interval_td / 3, "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+        pd.Timedelta,
+    )
+
+    check(assert_type(interval_i / 3.5, "pd.Interval[float]"), pd.Interval, float)
+    check(assert_type(interval_f / 3.5, "pd.Interval[float]"), pd.Interval, float)
+    check(
+        assert_type(interval_td / 3.5, "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+        pd.Timedelta,
+    )
+
+    check(assert_type(interval_i // 3, "pd.Interval[int]"), pd.Interval, int)
+    check(assert_type(interval_f // 3, "pd.Interval[float]"), pd.Interval, float)
+    check(
+        assert_type(interval_td // 3, "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+        pd.Timedelta,
+    )
+
+    check(assert_type(interval_i // 3.5, "pd.Interval[float]"), pd.Interval, float)
+    check(assert_type(interval_f // 3.5, "pd.Interval[float]"), pd.Interval, float)
+    check(
+        assert_type(interval_td // 3.5, "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+        pd.Timedelta,
+    )
+
+    check(assert_type(interval_i - 1, "pd.Interval[int]"), pd.Interval, int)
+    check(assert_type(interval_f - 1, "pd.Interval[float]"), pd.Interval, float)
+    check(
+        assert_type(interval_ts - pd.Timedelta(days=1), "pd.Interval[pd.Timestamp]"),
+        pd.Interval,
+        pd.Timestamp,
+    )
+    check(
+        assert_type(interval_td - pd.Timedelta(days=1), "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+        pd.Timedelta,
+    )
+
+    check(assert_type(interval_i - 1.5, "pd.Interval[float]"), pd.Interval, float)
+    check(assert_type(interval_f - 1.5, "pd.Interval[float]"), pd.Interval, float)
+    check(
+        assert_type(interval_ts - pd.Timedelta(days=1), "pd.Interval[pd.Timestamp]"),
+        pd.Interval,
+    )
+    check(
+        assert_type(interval_td - pd.Timedelta(days=1), "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+    )
+
+    check(assert_type(interval_i + 1, "pd.Interval[int]"), pd.Interval)
+    check(assert_type(interval_f + 1, "pd.Interval[float]"), pd.Interval)
+    check(
+        assert_type(interval_ts + pd.Timedelta(days=1), "pd.Interval[pd.Timestamp]"),
+        pd.Interval,
+    )
+    check(
+        assert_type(interval_td + pd.Timedelta(days=1), "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+    )
+
+    check(assert_type(interval_i + 1.5, "pd.Interval[float]"), pd.Interval)
+    check(assert_type(interval_f + 1.5, "pd.Interval[float]"), pd.Interval)
+    check(
+        assert_type(interval_ts + pd.Timedelta(days=1), "pd.Interval[pd.Timestamp]"),
+        pd.Interval,
+    )
+    check(
+        assert_type(interval_td + pd.Timedelta(days=1), "pd.Interval[pd.Timedelta]"),
+        pd.Interval,
+    )
```

Comment:
For all of the arithmetic operations, should test the reverse operator, e.g., `pd.Timedelta(days=1) + interval_td`

Reply:
Done, at least where allowed.